### DISCUSSION TITLE
MySQL CDC binlog change after server migration

### DISCUSSION THREAD

**Discussion by 0xffmeta** [apache/flink-cdc#2008] [Q&A]
Is there a way to change the binlog offset (e.g. xx timestamp) from the flink checkpoint? 

Backgroud is that we have a running flink job consume the MySQL CDC binlogs, but we have to migrate MySQL instance to another one which has a total diferent binlog offset. 

What's the best approach for this kind of migration? Do we allow the operator use the configed startup option even we want to restore from the flink checkpoint?

**Comment by cooltnt** [comment]
try to set
'scan.startup.mode' = 'timestamp',
'scan.startup.timestamp-millis' = 'timestamp(ms)'

timestamp: Skip snapshot phase and start reading binlog events from a specific timestamp.
https://github.com/ververica/flink-cdc-connectors/blob/master/docs/content/connectors/mysql-cdc.md#startup-reading-position
